### PR TITLE
Add diff3 support

### PR DESCRIPTION
**Pull request by wetneb**
For #496.

I have added test cases (failing) with some intuitive expectations of what I would find useful as a user. I haven't thought super hard about it so I expect some cases might need adjusting.

No idea if it can be implemented, but I guess if someone wants to have a go at this, those changes should be a good start and they would only need to figure out the clever part of the algorithm.

### CODE REVIEW

**Comment by slarse** [comment]
Hi @wetneb,

Sorry about the complete blackout in communications here, I've been busy with other things. I don't think there is any good reason to merge this before there's at least a POC of outputting the base revision for the structural merge, but I don't mind leaving the PR open for some time to see if anyone is interested in looking into it.

On that note, I outlined some potential solutions to outputting the base revision in https://github.com/ASSERT-KTH/spork/issues/496#issuecomment-1943668633.

**Comment by wetneb** [comment]
Hi @slarse,

I have a solution that almost ready, but it surfaced a bug in jgit's own implementation of diff3: https://github.com/eclipse-jgit/jgit/issues/38.
This PR is therefore blocked by that bug, for which I have a pending fix: https://eclipse.gerrithub.io/c/eclipse-jgit/jgit/+/1177976.
Once this is merged and released, this PR can then go ahead.

File: src/main/kotlin/se/kth/spork/spoon/pcsinterpreter/SporkTreeBuilder.kt [hunk 1]
```diff
@@ -263,25 +279,15 @@ internal class SporkTreeBuilder(
...
-                    }
-                    .findFirst()
-                if (predConflict.isPresent) {
-                    remainingInconsistencies.remove(predConflict.get())
-                    return nodes
-                }
+            val otherPred = delta.getOtherPredecessors(currentPcs).firstOrNull()
+            if (otherPred != null) {
+                remainingInconsistencies.remove(otherPred)
+                return nodes
```
Comment (wetneb) [inline src/main/kotlin/se/kth/spork/spoon/pcsinterpreter/SporkTreeBuilder.kt:290]: This is quite a simplification from the original so I suspect I might have overseen something here. Curious to know what you think.

**Review by wetneb — COMMENTED** [review]
Here is a working version, now that the fixed jgit has been released.

**Comment by monperrus** [comment]
@slarse ping?

**Comment by slarse** [comment]
@monperrus / @wetneb I don't think I will get around to integrating this. I've lost too much understanding of the internal workings of Spork over the years. I haven't written Java or Kotlin for years, either. I put a couple of hours going over this and trying to evaluate it, but I'm just not up for the task right now. [Given how my open source commitment has diminished in the past years](https://slar.se/taking-a-step-back-from-open-source-development.html), I don't think I'll ever be. My career is taking me other places.

@monperrus For the same reasons I stepped down from Spoon, I cannot put much time into maintaining Spork at this time. I can do small things, I can answer questions and provide what insights I remember, but for Spork to live on and keep growing, it needs a new maintainer. If you can staff one, feel free to treat the project as the sole property of the ASSERT group, I make no claims of ownership. But I feel I have done my part.

@wetneb I'm truly sorry to have kept you waiting all this time just to get back to you with "I can't do it". It must be disappointing. Trying to integrate this change a few months back, failing, and then completely sticking my head in the sand was quite the wake-up call for me. I hope @monperrus can find someone to integrate this, otherwise you can always fork Spork (hah!) and carry on.

**Comment by monperrus** [comment]
@slarse I fully understand. thanks for all the effort you put in Spoon and Spork.

**Comment by monperrus** [comment]
thanks a lot @wetneb, that's a super useful feature.

could you also document the new feature in the README? Thanks!

**Pull request merged**